src/hotspot/share/runtime/objectMonitor.inline.hpp

```diff
@@ -92,22 +92,22 @@ inline void ObjectMonitor::clear_common() {
   assert(contentions() <= 0, "must not be positive: contentions=%d", contentions());
   assert(_waiters == 0, "must be 0: waiters=%d", _waiters);
   assert(_recursions == 0, "must be 0: recursions=" INTX_FORMAT, _recursions);
-  assert(_object != NULL, "must be non-NULL");
+  assert(object() != NULL, "must be non-NULL");
 
   set_allocation_state(Free);
-  _object = NULL;
+  set_object(NULL);
 }
 
-inline void* ObjectMonitor::object() const {
-  return _object;
+inline oop ObjectMonitor::object() const {
+  return (oop)_object;
```

Comment:
`cast_to_oop(_object)` here?

Reply:
This function will change again with the next changeset (JDK-8247281)
so I'd rather stick with the C-style cast for this changeset.

Reply:
I kicked off a Mach5 Tier[1-3] and all the usual builds have passed.